feat: Implemented user authentication

MaGloCon opened the pull request:
- Added LoginView component to handle user login
- Added SignupView component to handle new user registration
- Integrated LoginView and SignupView in MainView
- Stored authenticated user and token in local storage
- Fetched movies only when user is authenticated
- Added logout functionality to clear user and token

MaGloCon committed:
feat: Implemented user authentication

- Added LoginView component to handle user login
- Added SignupView component to handle new user registration
- Integrated LoginView and SignupView in MainView
- Stored authenticated user and token in local storage
- Fetched movies only when user is authenticated
- Added logout functionality to clear user and token

Stanley-Okwii commented on src/components/main-view/main-view.jsx:
```diff
@@ -1,41 +1,64 @@
+    })
+      .then(response => response.json())
+      .then((data) => {
+        // console.log('Fetched data:', data);
```
While console.logs are good for debugging it's great to remove them when you are done.

MaGloCon committed:
Refactor user and token state initialization and user check

- Initialized user and token state with values from localStorage to persist user session
- Replace check for user existence with check for empty user object to handle cases where user state is an empty object
- Removed unnecessary console.log from data fetching function
- Updated logout button click handler to set user state to an empty object instead of null

Stanley-Okwii reviewed (approved):
This looks great. Let's get it merged 🎉.
